### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by ducbuianh**
### Title
[Chapter 9](https://www.railstutorial.org/book/advanced_login) of "Ruby on Rails Tutorial (Rails 5)" 

### Background
Training Rails

### Technical Changes
1. Add "Remember me" Checkbox
2. Add column "remember_token" to User table

### Test
Integration Test for User Remember function.

### Database Migration
Need to run db:migrate for new column added.

### CODE REVIEW

File: app/views/sessions/new.html.erb [hunk 1]
```diff
@@ -10,6 +10,11 @@
       <%= f.label :password %>
       <%= f.password_field :password, class: 'form-control' %>
 
+      <%= f.label :remember_me, class: 'checkbox inline' do %>
+        <%= f.check_box :remember_me %>
+        <span>Remember me on this computer</span>
```
Comment (haanhduclinh) [inline app/views/sessions/new.html.erb]: [IMO]
could you update to use I18n?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -3,18 +3,19 @@ def new
   end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user&.authenticate(params[:session][:password])
-      log_in user
-      redirect_to user_url(user)
+    @user = User.find_by(email: params[:session][:email].downcase)
+    if @user&.authenticate(params[:session][:password])
```
Comment (haanhduclinh) [inline app/controllers/sessions_controller.rb]: [Q]
do you know why we use `&` in this case ?

**Review by haanhduclinh — CHANGES_REQUESTED** [review]

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (ducbuianh) [reply app/controllers/sessions_controller.rb]: [A] This is Safe Navigation operator. I think this is a short way to check a nil variable before doing anything with it.

File: config/locales/en.yml [hunk 3]
```diff
@@ -38,6 +38,7 @@ en:
     login:
       title: Login
       btn_login: Log in
+      chkbox_remember_me: Remember me on this computer
```
Comment (haanhduclinh) [inline config/locales/en.yml]: [Q]You forgot to update I18n for Japan ?

**Review by haanhduclinh — CHANGES_REQUESTED** [review]

**Review by haanhduclinh — APPROVED** [review]
LGTM!

**Pull request merged**